src/main/java/uk/ac/cam/cl/dtg/segue/configuration/SegueGuiceConfigurationModule.java

```diff
@@ -914,7 +916,20 @@ private static SegueJobService getSegueJobService(final PostgresSqlDb database)
                     "SQL scheduled job that deletes expired reservations for the event booking system",
                     "0 0 7 * * ?", "db_scripts/scheduled/expired-reservations-clean-up.sql");
 
-            segueJobService = new SegueJobService(Arrays.asList(PIISQLJob, cleanUpOldAnonymousUsers, cleanUpExpiredReservations));
+            SegueScheduledJob syncMailjetUsers = new SegueScheduledSyncMailjetUsersJob(
+                    "syncMailjetUsersJob",
+                    "JavaJob",
+                    "Sync users to mailjet",
+                    "0 0 7,19 ? * * *");
+
+            segueJobService = new SegueJobService(Arrays.asList(PIISQLJob, cleanUpOldAnonymousUsers, cleanUpExpiredReservations, syncMailjetUsers));
+            
+            String mailjetKey = properties.getProperty(MAILJET_API_KEY);
+            String mailjetSecret = properties.getProperty(MAILJET_API_SECRET);
+
+            if (mailjetKey == null && mailjetSecret == null) {
```

Comment:
This would be much cleaner if the job was never added in the constructor in the first place, instead of trying to remove it if there is no key configured. But I think you are right that we do need to keep the removal of jobs, because Quartz retains jobs whether or not the API registers them and we don't want the MailJet job failing forever if (e.g. locally) it isn't wanted but was at some point registered.

Reply:
I believe this is addressed now and it runs fine with just the removal (no config set) even when there is no syncMailjetUsers job to be removed.